### PR TITLE
Fixes #16593: rl replace file headers to shorter spdx license identifier

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/16593

### CODE REVIEW

File: rudder-lang/src/ast/resource.rs [hunk 1]
```diff
@@ -1,32 +1,4 @@
...
-// the Software.
-//
-// Rudder is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-//
-// You should have received a copy of the GNU General Public License
-// along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+// SPDX-License-Identifier: GPL-3.0-only
```
Comment (amousset) [inline rudder-lang/src/ast/resource.rs]: I see in our license text:

```either version 3 of the License, or
// (at your option) any later version.```

So I guess it should be `GPL-3.0-or-later`


`

Comment (amousset) [reply rudder-lang/src/ast/resource.rs]: And I think we still need to keep the `Copyright` line

Comment (amousset) [reply rudder-lang/src/ast/resource.rs]: And we have an exception in our license that probably need to stay is some way.

File: rudder-lang/src/ast.rs [hunk 2]
```diff
@@ -1,32 +1,5 @@
...
-// the Software.
-//
-// Rudder is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-//
-// You should have received a copy of the GNU General Public License
-// along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+// SPDX-License-Identifier: GPL-3.0-only
```
Comment (amousset) [inline rudder-lang/src/ast.rs]: or later

File: rudder-lang/src/ast.rs [hunk 3]
```diff
@@ -1,32 +1,5 @@
...
+// SPDX-FileCopyrightText: © 2020 Normation SAS
```
Comment (amousset) [inline rudder-lang/src/ast.rs]: It should probably be "2019-2020"

Comment (amousset) [reply rudder-lang/src/ast.rs]: In general we need to keep the year the file was created

Comment (amousset) [reply rudder-lang/src/ast.rs]: And according to https://reuse.software/tutorial/ no need for "©"

File: rudder-lang/src/ast.rs (hunk 2 above)
Comment (gpoblon) [reply rudder-lang/src/ast.rs]: sorry, forgot about it. fixed

File: rudder-lang/src/ast.rs (hunk 3 above)
Comment (gpoblon) [reply rudder-lang/src/ast.rs]: fixed

**Comment by amousset** [comment]
Looks good to me.

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**